### PR TITLE
Remove vars

### PR DESCRIPTION
**Pull request by raphamorim**
Ref: #11766 #11752 #11776
3/3 Last one @gaearon 👍 

No more `vars` on `packages` source code :octocat:

### CODE REVIEW

File: packages/react-dom/src/client/ReactDOMFiberComponent.js [hunk 1]
```diff
@@ -887,10 +903,10 @@ export function diffHydratedProperties(
   assertValidProps(tag, rawProps, getStack);
 
   if (__DEV__) {
-    var extraAttributeNames: Set<string> = new Set();
```
Comment (gaearon) [inline packages/react-dom/src/client/ReactDOMFiberComponent.js]: Why was this extracted to top level? Looks like a local variable to me.

Comment (raphamorim) [reply packages/react-dom/src/client/ReactDOMFiberComponent.js]: hm, you right @gaearon. Updated :octocat:

**Comment by blling** [comment]
```
We could lint against it but this is a bit tricky because some files have to use var. I think ideally we should have a way to override lint configuration for a subset of files. We have the same problem in Prettier config (but solved it there).
```
If this is the last one , should we consider lint config as a follow up?

**Comment by gaearon** [comment]
We could but it’s not clear to me how this should work. Ideally we want to lint ES5 and ES6 code differently depending on the directory. Happy to discuss in a separate issue if you create one.

**Comment by jane42** [comment]
Nice

**Comment by blling** [comment]
@gaearon I have create one #11783

**Comment by blling** [comment]
I found `var` in some README.md docs, eg. [here](https://github.com/facebook/react/blame/master/packages/react/README.md#L15) and [here](https://github.com/facebook/react/blame/master/packages/react-dom/README.md#L16). 
Are they need to replace `var` with `const` ?

**Comment by gaearon** [comment]
That doesn't really matter. I'm fine leaving them as is for now.

**Pull request merged**